fixtures/get-platform-proxy-remote-bindings/tests/index.test.ts

```diff
@@ -15,123 +20,174 @@ const remoteWorkerName = `tmp-e2e-worker-test-remote-bindings-${randomUUID().spl
+				}>({
+					configPath: "./.tmp/wrangler.json",
+					experimental: { remoteBindings: true },
+				});
+
+				const response = await fetchFromWorker(env.MY_WORKER, "OK");
+				const workerText = await response?.text();
+				expect(workerText).toEqual(
+					"Hello from a remote Worker part of the getPlatformProxy remote bindings fixture!"
+				);
+
+				const kvValue = await env.MY_KV.get("test-key");
+				expect(kvValue).toEqual("remote-kv-value");
+
+				await dispose();
 			});
 
-			const workerText = await (
-				await env.MY_WORKER.fetch("http://example.com")
-			).text();
-			expect(workerText).toEqual(
-				"Hello from a remote Worker part of the getPlatformProxy remote bindings fixture!"
-			);
-
-			const kvValue = await env.MY_KV.get("test-key");
-			expect(kvValue).toEqual("remote-kv-value");
-
-			await dispose();
-		});
-
-		test("getPlatformProxy does not work with remote bindings if the experimental remoteBindings flag is not turned on", async () => {
-			const { env, dispose } = await getPlatformProxy<{
-				MY_WORKER: Fetcher;
-				MY_KV: KVNamespace;
-			}>({
-				configPath: "./.tmp/wrangler.json",
+			test("getPlatformProxy does not work with remote bindings if the experimental remoteBindings flag is not turned on", async () => {
+				const { env, dispose } = await getPlatformProxy<{
+					MY_WORKER: Fetcher;
+					MY_KV: KVNamespace;
+				}>({
+					configPath: "./.tmp/wrangler.json",
+				});
+
+				const response = await fetchFromWorker(
+					env.MY_WORKER,
+					"Service Unavailable"
+				);
+				const workerText = await response?.text();
+				expect(workerText).toEqual(
+					`[wrangler] Couldn\'t find \`wrangler dev\` session for service "${remoteWorkerName}" to proxy to`
+				);
+
+				const kvValue = await env.MY_KV.get("test-key");
+				expect(kvValue).toEqual(null);
+
+				await dispose();
 			});
-
-			const workerText = await (
-				await env.MY_WORKER.fetch("http://example.com")
-			).text();
-			expect(workerText).toEqual(
-				`[wrangler] Couldn\'t find \`wrangler dev\` session for service "${remoteWorkerName}" to proxy to`
-			);
-
-			const kvValue = await env.MY_KV.get("test-key");
-			expect(kvValue).toEqual(null);
-
-			await dispose();
-		});
-	});
+		},
+		{ timeout: 50_000 }
+	);
 } else {
 	test.skip("getPlatformProxy - remote bindings (no auth credentials)");
 }
 
+/**
+ * Tries to fetch from a worker multiple times until a response is returned which matches a specified
+ * statusText. Each fetch has a timeout signal making sure that it can't simply get stuck.
+ *
+ * This utility is used, instead of directly fetching from the Worker in order to prevent flakiness.
+ *
+ * @param worker The Worker to fetch from.
+ * @param expectedStatusText The response's expected statusText.
+ * @returns The successful Worker's response or null if no such response was obtained.
+ */
+async function fetchFromWorker(
+	worker: Fetcher,
+	expectedStatusText: string
+): Promise<Response | null> {
```

Comment:
I feel this can be a bit simpler:

```ts
async function fetchFromWorker(
	worker: Fetcher,
	expectedStatusText: string
): Promise<Response> {
  return vi.waitFor(async () => {
    const response = await worker.fetch("http://example.com", {signal: AbortSignal.timeout(5_000)});
    expect(response.status).toEqual(expectedStatusText);
    return response;
  });
}
```